src/pathwayvis/components/map/map.component.ts

```diff
@@ -314,6 +315,9 @@ class MapComponentCtrl {
 
         this.shared.decrement();
       });
+      infoPromise.then((data: any) => {
+        this._mapOptions.setMapInfo(data.data, id);
```

Comment:
This one is my favorite, but I do it everywhere unfortunately. One thing you could do to improve it a little is  ES6 object destructuring:

      infoPromise.then(({data:mapInfo}: any) => {
        this._mapOptions.setMapInfo(mapInfo, id);